packages/host/app/services/local-realm.ts

```diff
@@ -132,22 +132,25 @@ export default class LocalRealm extends Service {
       this.state = { type: 'fastboot', worker: undefined };
       return;
     }
-    if (demoRealmURL) {
+    if (!localRealmEnabled) {
       this.state = {
-        type: 'demo-realm',
+        type: 'remote-realm',
         worker: undefined,
       };
       return;
     }
     await Promise.resolve();
+    let indexResponse = await fetch('./index.html');
+    let indexHTML = await indexResponse.text();
     this.state = { type: 'checking-worker' };
     let worker = await this.ensureWorker();
     this.state = {
       type: 'requesting-handle',
       worker,
       response: new Deferred<DirectoryHandleResponse>(),
+      indexHTML,
     };
-    send(this.state.worker, { type: 'requestDirectoryHandle' });
+    send(this.state.worker, { type: 'requestDirectoryHandle', indexHTML });
```

Comment:
maybe we can send the URL instead and the worker can just do the fetch of the index.html itself